packages/core/src/binary-data/object-store/object-store.service.ee.ts

```diff
@@ -243,28 +220,46 @@ export class ObjectStoreService {
 
 	private async request<T>(
 		method: Method,
-		host: string,
 		rawPath = '',
 		{ qs, headers, body, responseType }: RequestOptions = {},
 	) {
-		const path = this.toPath(rawPath, qs);
+		const {
+			host,
+			bucket: { region },
+		} = this.s3Config;
+		let url = this.baseUrl;
+		if (rawPath && rawPath !== '/') {
+			url = `${url}/${rawPath}`;
+		}
+		if (qs) {
+			url +=
+				'?' +
+				Object.entries(qs)
+					.map(([key, value]) => `${key}=${value}`)
+					.join('&');
+		}
+		const path = new URL(url).pathname;
```

Comment:
Would it be cleaner to create an instance of URL class with baseUrl, and set the `pathname` and `query` instead of building a string and then parsing it with URL?

Reply:
Also I liked that this was its own method. Now it makes this method more messy